### ISSUE TITLE
failing to set auth header for elasticsearch in 7.14.0

### ISSUE THREAD

**Issue by kares** [elastic/logstash#13122]
LS 7.14.0 updated its `elasticsearch` gem dependency to (`-> 7`) **7.5.0** which leads to a regression with the filter plugin when (user/password) credentials are used. This also affects any other authorization mechanism such as `cloud_auth` and `api_key` as the plugin passes all of those as a header to `Elasticsearch::Client`.
The plugin is setting the [Authorization header with `transport_options: { headers: ... }`](https://github.com/logstash-plugins/logstash-filter-elasticsearch/blob/v3.9.3/lib/logstash/filters/elasticsearch/client.rb#L21) but this isn't working properly:
```
[logstash.filters.elasticsearch][main] New ElasticSearch filter client {:hosts=>["http://localhost:9200"]}
[ERROR][logstash.javapipeline    ][main] Pipeline error {:pipeline_id=>"main", :exception=>#<Elasticsearch::Transport::Transport::Errors::Unauthorized: [401] >
```

- [x] elasticsearch filter reproducer: https://github.com/logstash-plugins/logstash-filter-elasticsearch/issues/147
- [x] elasticsearch input is also affected https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/153

*NOTE: LS has been using elasticsearch gem version 5.0.5 for long time (through all 7.x series and even before).*

**Comment by nicenemo** [comment]
Not only the filter plugin. also the input plugin for elasticsearch.

**Comment by nicenemo** [comment]
The filter plugin already had that  problem with 7.13.x for api_keys.

**Comment by nicenemo** [comment]
Related to: https://github.com/logstash-plugins/logstash-filter-elasticsearch/issues/129 ?

**Comment by kares** [comment]
Thanks Hans, it isn't really related to https://github.com/logstash-plugins/logstash-filter-elasticsearch/issues/129 
as noted in the description this is due a client bug.

**Comment by kares** [comment]
LS team pushed fixes for the ES plugins that got affected by this regression, users on LS 7.14.0 are advised to update:
- `bin/logstash-plugin update logstash-filter-elasticsearch` (should update to version >= 3.9.4)
- `bin/logstash-plugin update logstash-input-elasticsearch` (should update to version >= 4.9.2)

the updated gems are expected to be available in LS **>= 7.14.1**

**Comment by JohannesMahne** [comment]
Just for users finding this solution, but get `403` issues while updating the plugins:

```
Error Net::HTTPServerException, retrying 1/10
403 "Forbidden"
```

This can be because access to `https://rubygems.org` might be restricted by a proxy.

Please see our [documentation on managing offline plugins](https://www.elastic.co/guide/en/logstash/current/offline-plugins.html).

**Comment by kares** [comment]
closing - the updated plugins are available in LS >= 7.14.1
